### DISCUSSION TITLE
wasm-merge call_indirect

### DISCUSSION THREAD

**Discussion by spotandjake** [WebAssembly/binaryen#5782] [Q&A] [answered]
with `wasm-merge` because it uses mutliple tables, there is no way for first class functons based on `call_indirect` to pass the module boundary, as wasm's call_indirect table index is static. Im wondering if there has been any thought into this. its not really a problem when talking about linking wasm generated from hand by wat but it would make languages with first class functions that want to take advantage of wasm-merge's life a lot harder.

**Comment by tlively** [comment, marked as answer]
Here's the scheme I would expect such languages to follow: If module `A` wants to `call_indirect` a function from module `B`, then `A` would import the function from `B` and put it in `A`'s table. Separately, module `B` might also put the function in `B`'s table for its own use. When modules `A` and `B` are merged, the function would still be placed in both tables and all the `call_indirect`s would still work.
